src/Http/Owin/src/OwinEnvironment.cs

```diff
@@ -478,4 +399,218 @@ public FeatureMap(Func<TFeature, object> getter, Func<object> defaultFactory, Ac
         {
         }
     }
+
+    private sealed class OwinEntries : IEnumerable<KeyValuePair<string, FeatureMap>>
+    {
+        private static readonly IDictionary<string, FeatureMap> _entries = new Dictionary<string, FeatureMap>
```

Comment:
if immutable dictionary is available, I wonder if we should use that instead, to advertise 100% "this will not change"; alternatively (other than for the deferred per-context populate, which might need thought), I wonder if this could just be a `switch` expression...

Reply:
changed type of `_entries` to `IReadonlyDictionary<string, FeatureMap>` + creating the `_entries` as `ImmutableDictionary` to completely make sure we are not breaking the shared-state dictionary.